cogs/wordfilter/wordfilter.py

```diff
@@ -639,14 +639,14 @@ def _filterWord(words, string):
         return re.sub(regex, _censorMatch, string, flags=re.IGNORECASE)
 
 
-def _isOneWord(string):
+def _isOneWord(string: str):
     return len(string.split()) == 1
 
 
-def _isAllFiltered(string):
+def _isAllFiltered(string: str):
     words = string.split()
     cnt = 0
     for word in words:
-        if bool(re.search("[*]+", word)):
+        if bool(re.search(r"^\*+$", word, flags=re.MULTILINE)):
```

Comment:
Since doing RegEx is usually more expensive than basic string checks (although in this case word length is not huge), we can just check if everything is `*`.
```suggestion
        if all(map(lambda c: c == "*", word)):
```
Also, in case we are concerned with a string from its beginning to its end, we can use [`re.fullmatch()`](https://docs.python.org/3/library/re.html#re.fullmatch) as well. This is less in complexity compared to [`re.search()`](https://docs.python.org/3/library/re.html#re.search), but either way, I don't see the need for RegEx here.

Reply:
Yep, that's a good point.